src/GafferScene/ShaderPlug.cpp

```diff
@@ -48,31 +48,23 @@ using namespace Gaffer;
 using namespace GafferScene;
 
 //////////////////////////////////////////////////////////////////////////
-// Internal utilities
+// ShaderPlug
 //////////////////////////////////////////////////////////////////////////
 
-namespace
-{
+IE_CORE_DEFINERUNTIMETYPED( ShaderPlug );
 
-const GafferScene::Shader *shader( const ShaderPlug *plug )
+
+const Gaffer::Plug *ShaderPlug::getShaderOutPlug() const
```

Comment:
Sure. I've slowly been getting into the habit of keeping as much as possible out of the public headers, but for a non-virtual private method with a single argument equivalent to `this`, I guess that is contrived. One thing though - `get*()` methods in Gaffer are always paired with a `set*()` equivalent, so this should just be `shaderOutPlug()`...